src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASED_ARRAY_SIZE = 10;
+    private T[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = (T[]) new Object[BASED_ARRAY_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Out of bout exception");
+        }
+        grow();
+        System.arraycopy(objects,index,objects,index + 1,objects.length - index - 1);
+        objects[index] = value;
+        size++;
+    }
 
+    private void grow() {
+        if (size == objects.length) {
+            T[] arraySours;
+            arraySours = objects;
+            objects = (T[])new Object[arraySours.length + 5];
+            System.arraycopy(arraySours,0,objects,0,objects.length - 5);
+
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        int size = this.size;
+        int listIndex = 0;
+        for (int j = size; j < list.size() + size; j++) {
+            objects[j] = list.get(listIndex);
+            listIndex++;
+            this.size++;
+        }
```

Comment:
what about grow method? What if we need to increase the initial array to add the whole list? Maybe we can just use our `add(T value)`  method for every element in the incoming list?